internal/engine/upper.go

```diff
@@ -131,40 +133,48 @@ func (u Upper) CreateManifests(ctx context.Context, manifests []model.Manifest,
 			case <-ctx.Done():
 				return ctx.Err()
 			case event := <-sw.events:
-				oldState := buildStates[event.manifest.Name]
-				buildState := oldState.NewStateWithFilesChanged(event.files)
-				buildStates[event.manifest.Name] = buildState
-
-				spurious, err := buildState.OnlySpuriousChanges()
-				if err != nil {
-					logger.Get(ctx).Infof("build watch error: %v", err)
-				}
+				if eventContainsConfigFiles(event) {
+					t, err := tiltfile.Load(tiltfile.FileName, os.Stdout)
+					if err != nil {
+						// TODO(dmiller) should we fail here, or is this OK?
+						return err
```

Comment:
i don't think we should fail here. we kind of expect that people will make mistakes writing a tiltfile

we probably should keep some state variable that the Tiltfile needs to be reloaded, and only reset that state variable when the tiltfile successfully loads. if it doesn't load, that just counts as a failed build

Reply:
Agreed, good idea on that extra piece of state.

Reply:
what about the next loop iteration?